maistra.mk

```diff
@@ -1,3 +1,63 @@
+# Container images labels
+MAISTRA_VERSION ?= 2.2.0
+ISTIO_VERSION ?= 1.12.7
+
+# Maistra specific vars
+MAISTRA_IMAGES ?= maistra-image.pilot maistra-image.proxy maistra-image.cni
+
+# Building container image for specific Maistra component
+MAISTRA_IMAGE_RULE ?= time (mkdir -p $(DOCKER_BUILD_TOP)/$@ && TARGET_ARCH=$(TARGET_ARCH) ./tools/docker-copy.sh $^ $(DOCKER_BUILD_TOP)/$@ && cd $(DOCKER_BUILD_TOP)/$@ $(BUILD_PRE) && $(CONTAINER_CLI) build $(BUILD_ARGS) -t $(HUB)/$(subst maistra-image.,,$@):$(TAG) -f Dockerfile$(suffix $@) . )
+
+.PHONY: maistra-image.pilot
+maistra-image.pilot: BUILD_PRE=&& chmod 644 envoy_bootstrap.json gcp_envoy_bootstrap.json
+maistra-image.pilot: BUILD_ARGS=--build-arg MAISTRA_VERSION=${MAISTRA_VERSION} --build-arg ISTIO_VERSION=${ISTIO_VERSION}
+maistra-image.pilot: ${ISTIO_ENVOY_BOOTSTRAP_CONFIG_DIR}/envoy_bootstrap.json
+maistra-image.pilot: ${ISTIO_ENVOY_BOOTSTRAP_CONFIG_DIR}/gcp_envoy_bootstrap.json
+maistra-image.pilot: $(ISTIO_OUT_LINUX)/pilot-discovery
+maistra-image.pilot: $(ISTIO_OUT_LINUX)/mec
+maistra-image.pilot: maistra/Dockerfile.pilot
+	$(MAISTRA_IMAGE_RULE)
+
+.PHONY: maistra-image.proxy
+maistra-image.proxy: BUILD_PRE=&& chmod 644 envoy_bootstrap.json gcp_envoy_bootstrap.json
```

Comment:
we should perhaps add something like
```
maistra-image.proxy: VERSION=${MAISTRA_VERSION}
```
on the very beginning of these dependencies so that we get the right version in the output of `<binary> version` subcommand. Example:

Before:
```
docker run --rm -it quay.io/jwendell/proxy:2.2.0 version
version.BuildInfo{Version:"1.12-dev", GitRevision:"c30e9344bb503d749f9936e82b39706ed9d54730", GolangVersion:"go1.17.9", BuildStatus:"Clean", GitTag:"1.12.6-113-gc30e9344bb"}
```

After:
```
$ docker run --rm -it quay.io/jwendell/proxy:2.2.0 version
version.BuildInfo{Version:"2.2.0", GitRevision:"c30e9344bb503d749f9936e82b39706ed9d54730-dirty", GolangVersion:"go1.17.9", BuildStatus:"Modified", GitTag:"1.12.6-113-gc30e9344bb"}
```

Same to pilot and cni.

Reply:
```
$ docker run --rm -it quay.io/bmangoen/pilot-ubi8:2.2.0 version
version.BuildInfo{Version:"2.2.0", GitRevision:"69a592bccfa1bb3c620286a8840db4c51dcdf2ab-dirty", GolangVersion:"go1.17.9", BuildStatus:"Modified", GitTag:"1.7.0-alpha.0-4267-g69a592bccf"}

$ docker run --rm -it quay.io/bmangoen/istio-cni-ubi8:2.2.0 install-cni version
version.BuildInfo{Version:"2.2.0", GitRevision:"69a592bccfa1bb3c620286a8840db4c51dcdf2ab-dirty", GolangVersion:"go1.17.9", BuildStatus:"Modified", GitTag:"1.7.0-alpha.0-4267-g69a592bccf"}

$ docker run --rm -it quay.io/bmangoen/proxyv2-ubi8:2.2.0 version
version.BuildInfo{Version:"2.2.0", GitRevision:"69a592bccfa1bb3c620286a8840db4c51dcdf2ab-dirty", GolangVersion:"go1.17.9", BuildStatus:"Modified", GitTag:"1.7.0-alpha.0-4267-g69a592bccf"}
```